### ISSUE TITLE
request count over the limit

### ISSUE THREAD

**Issue by AugustBurnsRed** [php-tmdb/api#96]
Hi, I have a autocomplete field where I search for movies by title. When I have a lot of result I often get this error: Your request count (41) is over the allowed limit of 40.

How can I limit the result? Right now my query is : 

$output =  \Tmdb::getSearchApi()->searchMovies($name,array('language' => 'fr'));

Thanks

**Comment by wtfzdotnet** [comment]
This should be catched by an mechanism, that delays the upcoming requests. I've noticed this not working as it should either, little short on time right now but I'll try and take a look at this around this weekend.

If you feel like looking at it, review the 427 pull request from a couple of weeks ago. This should postpone and block the current request until the sleep time was reached. It is however not working as I expected it should. Need to cover this with some proper tests.

I think I'll contact Travis about this though, for any site that has only slight traffic this will become a problem.

Make sure caching works on your end, so this won't happen in other areas as much.

**Comment by wtfzdotnet** [comment]
Secondly, for auto complete functionality you should also provide the search_type parameter with the value ngram

See http://docs.themoviedb.apiary.io/

**Comment by wtfzdotnet** [comment]
I just posted a question to Travis, trying to figure out what the rate limits are on ngram queries.

**Comment by wtfzdotnet** [comment]
```
Hi +Michael Roterman 

There's a few apps in the wild that I know of that use autocomplete, TodoMovies and Moovee for example. I use these apps fairly often and have never noticed the rate limits to cause any grief. As always though, since the rate limits are IP based, should someone be proxying requests through a server (single IP) then yes, you would absolutely run into issues quite easily.

As it stands right now, the rate limits are the same for every request API wide.﻿
```

I think the best idea for auto completion is to make a separate JavaScript implementation, that makes use of guest sessions locally. The limits are tied per external ip.

**Comment by AugustBurnsRed** [comment]
Hi, thanks for your answers. But my problem is not when I do a lot of request, it's when I do a request with a common name. 

If search something like 'man' : Tmdb::getSearchApi()->searchMovies('man',array('language' => 'fr', 'search_type' => 'ngram'));

It would return me the: Your request count is over the allowed limit of 40. I get this error just once per search term because after that it is cache

**Comment by wtfzdotnet** [comment]
Rate limiting is fixed, d8c4339e4411d6feebeb15c6a70b2a224fa1e61f
